open-craft/edx-platform
# feat: System defined taxonomies

## Description

Add the object tags for the system defined taxonomies that will be used in the platform. In addition, a migration has been created that loads the fixtures that create the System defined taxonomies

## Testing instructions

- Get the `master` [openedx/devstack](https://github.com/openedx/devstack) up and running.
- Pull this branch into edx-platform
- Install dependencies and update migrations by running `paver install_prereqs` in the LMS shell (from devstack dir, run `make lms-shell`) and CMS shell (`make studio-shell`).
- Run the new migrations to create the System defined Taxonomies by running `./manage.py lms migrate` in the LMS shell.
- Open a shell with `./manage.py lms shell`
- Check the new created System-defined taxonomies:
```
>>> from openedx_tagging.core.tagging.models import Taxonomy
>>> Taxonomy.objects.filter(id__lt=0)
<QuerySet [<ContentOrganizationTaxonomy> (-3) Content Authors, <ContentAuthorTaxonomy> (-2) Organizations, <ContentLanguageTaxonomy> (-1) Languages]>
>>> lang = Taxonomy.objects.get(id=-1)
>>> lang
<ContentLanguageTaxonomy> (-1) Languages
>>> lang.tag_set.all()
```
- Revert the migration with `./manage.py lms migrate content_tagging 0001`
- Check that System-defined taxonomies are deleted:
```
>>> from openedx_tagging.core.tagging.models import Taxonomy, Tag
>>> Taxonomy.objects.filter(id__lt=0)
<QuerySet []>
>>> Tag.objects.all()
<QuerySet []>
```

## Deadline

None

## Other information

Relates to https://github.com/openedx/modular-learning/issues/77

Depends on https://github.com/open-craft/openedx-learning/pull/3

Data migrations added here can be easily rolled back.
